src/pages/index.js

```diff
@@ -3,104 +3,248 @@ import FormValidator from "../components/FormValidator.js";
+  section[method](cardElement);
 }
 
 function getCardElement(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    (cardInstance) => openConfirmPopup(cardInstance),
+    handleImageClick,
+    handleLikeButton
+  );
   return card.getView();
 }
 
 function handleImageClick(data) {
-    popupWithImage.open({ name: data.name, link: data.link });
+  console.log(data.name);
+  popupWithImage.open({ name: data.name, link: data.link });
+}
+
+function handleLikeButton(cardId, likeButton) {
+  if (!cardId) return;
+  const isLiked = likeButton.classList.contains("card__like-button_active");
+
+  api
+    .toggleLike(cardId, isLiked)
+    .then((updatedCard) => {
+      if (updatedCard && "isLiked" in updatedCard) {
+        likeButton.classList.toggle(
+          "card__like-button_active",
+          updatedCard.isLiked
+        );
+      }
+    })
+    .catch((error) => console.error("Error updating like:", error));
 }
 
 /*Event Handler*/
 
-function handleProfileEditSubmit(inputValue) {
-    userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+function handleProfileEditSubmit(event) {
+  const inputValues = popupWithEditProfileForm.getInputValues(); // Get form values
+
+  if (!inputValues.name || !inputValues.about) {
+    return;
+  }
+
+  popupWithEditProfileForm.setLoadingText(true);
+
+  api
+    .updateUserInfo(inputValues.name, inputValues.about)
+    .then((updatedUser) => {
+      userInfo.setUserInfo(updatedUser);
+      popupWithEditProfileForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditProfileForm.setLoadingText(false);
+    });
+}
+
+function handleAvatarUpdateSubmit(event) {
+  const inputvalues = popupWithEditAvatarForm.getInputValues(); // Get form values
+  const avatarUrl = inputvalues.avatar;
+
+  if (!avatarUrl) {
+    return;
+  }
+
+  popupWithEditAvatarForm.setLoadingText(true);
+  api
+    .updateProfileAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      popupWithEditAvatarForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditAvatarForm.setLoadingText(false);
     });
-    popupWithEditProfileForm.close();
-    editFormValidator.disableSubmitButton();
 }
 
 function handleAddCardFormSubmit(inputValue) {
-    const cardData = {
-        name: inputValue.title,
-        link: inputValue.url,
-    };
-    renderCard(cardData);
-    popupWithAddCardForm.close();
-    addCardForm.reset();
-    addFormValidator.disableSubmitButton();
+  const cardData = {
+    name: inputValue.title,
+    link: inputValue.url,
+  };
+
+  popupWithAddCardForm.setLoadingText(true);
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      renderCard(newCard);
+    })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**